CMakeLists.txt

```diff
@@ -160,7 +160,7 @@ else()
   set(THREADS_PREFER_PTHREAD_FLAG ON)
   set(CMAKE_THREAD_PREFER_PTHREAD ON)
   find_package(Threads REQUIRED)
-  add_cxx_flag("-std=c++14")
+  add_cxx_flag("-std=c++17")
```

Comment:
Isn't there a builtin CMake command to set the C++ standard?

Reply:
https://cmake.org/cmake/help/latest/prop_tgt/CXX_STANDARD.html and https://cmake.org/cmake/help/latest/prop_tgt/CXX_STANDARD_REQUIRED.html#prop_tgt:CXX_STANDARD_REQUIRED

Reply:
Might be worth doing that transition separately as there be a history as to why are not currently using that.